docs/installation/prepare/microcontrollers/_index.md

```diff
@@ -1,387 +1,353 @@
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+cargo install cargo-espflash v2.0.0-rc.1
+```
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+cargo install ldproxy
+```
+
+To download and install the esp-rs toolchain, run:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+espup install -s -f ~/esp/export-rs.sh -v 1.67.0
+```
+
+To activate the ESP Rust toolchain, run the following command to source (`.`) the activation script `export-rs.sh`:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+. $HOME/esp/esp-idf/export-rs.sh
+```
+
+To avoid conflicts with other toolchains, adding this command to your `.bashrc` or `.zshrc` is not recommended.
+Save this command to run in any future terminal session where you need to activate the ESP Rust development framework.
+
+## Install the Micro-RDK
+
+### Create a New Robot
+
+Navigate to [the Viam app](https://app.viam.com) and [add a new robot](/manage/fleet/robots/#add-a-new-robot) in your desired location.
+
+* Click on the name of the robot to go to the robot's page.
+* Skip the instructions in the [**Setup** tab](/manage/fleet/robots/#setup), as the setup instructions there are not for microcontrollers.
+* Keep your `Mode` and `Architecture` selections at default.
+
+### Configure an `esp32` Board
+
+{{< alert title="Info" color="info" >}}
+The`esp32` [board](/components/board/) model is not currently provided for you as an option in [the Viam app](https://app.viam.com), so you cannot use the **Config Builder** to configure this board.
+{{< /alert >}}
+
+To add an `esp32` board, navigate to the **Config** tab of your robot's page in [the Viam app](https://app.viam.com) and select **Raw JSON** mode.
+
+Copy the following JSON template and paste it into your configuration:
+
+{{< tabs name="Configure an esp32 Board" >}}
+{{% tab name="JSON Template"%}}
+
+```json
+{
+  "attributes": {
+    "pins": [
+      <int>
+    ],
+    "analogs": [
+      {
+        "pin": "<number>", 
+        "name": "<your-analog-name>"
+      }
+    ]
+  },
+  "depends_on": [],
+  "model": "esp32",
+  "name": "<your-board-name>",
+  "type": "board"
+}
+```
+
+{{% /tab %}}
+{{% tab name="JSON Example" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "attributes": {
+    "pins": [
+      15
+    ],
+    "analogs": [
+      {
+        "pin": "34",
+        "name": "sensor"
+      }
+    ]
+  },
+  "depends_on": [],
+  "model": "esp32",
+  "name": "board",
+  "type": "board"
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+Edit and fill in the attributes as applicable.
+Click **Save config**.
+The following attributes are available for `esp32` boards:
+
+| Name | Type | Inclusion | Description |
+| ---- | ---- | --------- | ----------- |
+| `pins` | object | Required | The {{< glossary_tooltip term_id="pin-number" text="pin number" >}} of any GPIO pins you wish to use as input/output with the [`GPIOPin` API](/program/apis/#gpio-pins). |
```

Comment:
@npmenard Is this only required if accessing pins through the Board API? or do entries also need to be here for if a motor is being configured on these gpio pins?

Reply:
only for pins accessed as GPIO